pkg/collector/reconcile/service.go

```diff
@@ -48,6 +50,14 @@ func Services(ctx context.Context, params Params) error {
 		}
 	}
 
+	if params.Instance.Spec.TargetAllocator.Enabled {
+		_, err := GetPromConfig(params)
```

Comment:
This is actually for retrieving the Prometheus part of the config. The deployment and the service use it for validation such that they do not execute if a invalid Prometheus is received. The ConfigMap uses it to actually retrieve the config and send it to the target allocator.

Reply:
In that case, it is fine to call it for the configmap, but validation should generally be done in the admission webhook.

Reply:
It is a validation for checking if a Prometheus config exists or not. The check was added so that the deployment/configmap resource is not created if the target allocation option is enabled but there is no Prometheus config present.
Adding this validation in the admission webhook may not be right since the collectors should still spawn even if a Prometheus config is not present. This logic should not control anything else other than the target allocator.